pkg/plugin/util/helpers.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/config"
+)
+
+// IsOnlyCRDVersion returns true if crdVersion is the first or the only existing CRD version in the project.
+func IsOnlyCRDVersion(config config.Config, crdVersion string) bool {
+	return isOnlyAPIVersion(config.ListCRDVersions(), crdVersion)
+}
+
+// IsOnlyWebhookVersion returns true if webhookVersion is the first or the only existing webhook version in the project.
+func IsOnlyWebhookVersion(config config.Config, webhookVersion string) bool {
+	return isOnlyAPIVersion(config.ListWebhookVersions(), webhookVersion)
+}
+
+func isOnlyAPIVersion(versions []string, version string) bool {
+	return len(versions) == 0 || (len(versions) == 1 && versions[0] == version)
+}
```

Comment:
I understand that you change that for a List of webhook versions and List of CRD versions to allow the possibility of the tool works with many versions in the future. 

However, I do not think that we should have IsOnly funcs here. We would still have isWebhoohVersionCompatible  and IsCRDVersionCompatible where its responsibility would ensure that the version informed can or not be used.

Reply:
We talked about this and `hasDifferentAPIVersion` seems like a better name (for the negated function).